### PR TITLE
fix(security): removed potential backdoor and fixed test

### PR DESCRIPTION
**Pull request by Jayclifford345**
This pull request removes support for the global `__PathfinderTestingMode` flag, which previously allowed bypassing security checks for URL validation in tests and debug panels. The code now enforces stricter security policies for content fetching and rendering, ensuring that only trusted sources are allowed, even in test environments. The tests have been updated to reflect these changes, and references to the testing mode have been removed throughout the codebase.

**Security enforcement improvements:**

* Removed all logic that bypassed HTTPS and domain validation when `__PathfinderTestingMode` was enabled in `content-fetcher.ts`, `content-renderer.tsx`, and related helpers. Now, only dev mode and localhost are allowed as exceptions, not test mode. [[1]](diffhunk://#diff-fdb7f7c27090da3f5e37c5391ec999608b1d007a150f8ceae2805b9b0e022172L35-L43) [[2]](diffhunk://#diff-fdb7f7c27090da3f5e37c5391ec999608b1d007a150f8ceae2805b9b0e022172L87-R81) [[3]](diffhunk://#diff-fdb7f7c27090da3f5e37c5391ec999608b1d007a150f8ceae2805b9b0e022172L370-R360) [[4]](diffhunk://#diff-2f5d1484bb713811dc2787faa6c17544c56649413bf1c21856ab8981c6281159L211-R211)
* Updated security-related tests in `security-mitm.test.ts` and `content-fetcher.test.ts` to remove all cases and setup/teardown logic related to the testing mode flag. Tests now validate security enforcement without bypasses. [[1]](diffhunk://#diff-8726088620859951843b72bbd596a8be572c7731a445fd003e9beabef40d1797L33) [[2]](diffhunk://#diff-8726088620859951843b72bbd596a8be572c7731a445fd003e9beabef40d1797L72-L84) [[3]](diffhunk://#diff-8726088620859951843b72bbd596a8be572c7731a445fd003e9beabef40d1797L107) [[4]](diffhunk://#diff-8726088620859951843b72bbd596a8be572c7731a445fd003e9beabef40d1797L315) [[5]](diffhunk://#diff-8726088620859951843b72bbd596a8be572c7731a445fd003e9beabef40d1797L373) [[6]](diffhunk://#diff-8726088620859951843b72bbd596a8be572c7731a445fd003e9beabef40d1797L410-L454) [[7]](diffhunk://#diff-b830af024c52f999f4b9e5989357c77286242b6524e664a0552887ae072bad00L16-L20) [[8]](diffhunk://#diff-b830af024c52f999f4b9e5989357c77286242b6524e664a0552887ae072bad00L68-L79)

**Test suite improvements:**

* Clarified comments in `context-security.test.ts` to explain the use of static domain lists for allowlist/denylist tests, avoiding coupling to implementation details. [[1]](diffhunk://#diff-7fb120ec4bdbc03a29785609c8b40f3dd6f88bf125034de7ee3398d862a7380bR6-R11) [[2]](diffhunk://#diff-7fb120ec4bdbc03a29785609c8b40f3dd6f88bf125034de7ee3398d862a7380bR177-R178)
* Updated valid and invalid domain lists in recommender service URL validation tests to match production allowlist rules and explicitly handle edge cases. [[1]](diffhunk://#diff-7fb120ec4bdbc03a29785609c8b40f3dd6f88bf125034de7ee3398d862a7380bR204-R209) [[2]](diffhunk://#diff-7fb120ec4bdbc03a29785609c8b40f3dd6f88bf125034de7ee3398d862a7380bR249-R255)

### CODE REVIEW

File: src/utils/context/context-security.test.ts [hunk 1]
```diff
@@ -193,9 +201,12 @@ describe('Security: Recommender Service URL Validation', () => {
         searchParams: {},
       };
 
+      // Static domain list for testing - should match production allowed domains
+      // recommender.grafana.com and recommender.grafana-dev.com are the allowed domains
       const validUrls = [
         'https://recommender.grafana.com/recommend',
-        'https://recommender-staging.grafana.com/recommend',
+        'https://recommender.grafana-dev.com/recommend',
+        'https://api.recommender.grafana.com/recommend', // Subdomain of allowed domain
```
Comment (Jayclifford345) [inline src/utils/context/context-security.test.ts]: oh wow, this is getting worse. will fix

**Review by jdbaldry — APPROVED** [review]
Thanks for cleaning this up
